src/package/packageVersionCreate.ts

```diff
@@ -649,16 +650,19 @@ export class PackageVersionCreate {
         if (!packageName) throw messages.createError('errorMissingPackage', [this.options.packageId]);
       }
       packageObject = this.project.findPackage(
-        (namedPackageDir) => namedPackageDir.package === packageName || namedPackageDir.name === packageName
+        (namedPackageDir) =>
+          (isPackagingDirectory(namedPackageDir) && namedPackageDir.package === packageName) ||
```

Comment:
I don't think this logic matches, should it be 
`isPackagingDirectory(namedPackageDir) &&(namedPackageDir.package === packageName || namedPackageDir.name === packageName)`?